### PR TITLE
datatype: add decsription for date time and string

### PR DESCRIPTION
**Pull request by lamxTyler**

### CODE REVIEW

File: sql/datatype.md [hunk 1]
```diff
@@ -97,10 +98,88 @@ TODO
...
+> 类似于 BLOB, 区别在于最大列长度为 255。
+
+MEDIUMBLOB
+> 类似于 BLOB, 区别在于最大列长度为 16777215。
+
+LONGBLOB
+> 类似于 BLOB, 区别在于最大列长度为 4294967295。
+
+TEXT[(M)] [CHARACTER SET charset_name] [COLLATE collation_name]
+> 文本串。M 表示最大列长度，范围是0到65535。VARCHAR 的最大实际长度由最长的行的大小和使用的字符集确定。
```
Comment (andelf) [inline sql/datatype.md]: s/VARCHAR/TEXT/

File: sql/datatype.md [hunk 2]
```diff
@@ -97,10 +98,88 @@ TODO
...
+> 类似于 TEXT, 区别在于最大列长度为 255。
+
+MEDIUMTEXT [CHARACTER SET charset_name] [COLLATE collation_name]
+> 类似于 TEXT, 区别在于最大列长度为 16777215。
+
+LONGTEXT [CHARACTER SET charset_name] [COLLATE collation_name]
+> 类似于 TEXT, 区别在于最大列长度为 4294967295。
+
+ENUM('value1','value2',...) [CHARACTER SET charset_name] [COLLATE collation_name]
+> 枚举。只能有一个值的字符串对象，其值通常选自允许值列表中，在某些情况下也可以是空串或者 NULL。
```
Comment (andelf) [inline sql/datatype.md]: different meaning from:

> An ENUM is a string object with a value chosen from a list of permitted values that are enumerated explicitly in the column specification at table creation time.

Comment (lamxTyler) [reply sql/datatype.md]: `An enumeration. A string object that can have only one value, chosen from the list of values 'value1', 'value2', ..., NULL or the special '' error value.` ?

Comment (andelf) [reply sql/datatype.md]: never mind. that's ok

File: sql/datatype.md [hunk 3]
```diff
@@ -97,10 +98,88 @@ TODO
...
+> 日期和时间的组合。支持的范围是`1000-01-01 00:00:00.000000`到`9999-12-31 23:59:59.000000`。
+以`YYYY-MM-DD HH:MM:SS[.fraction]`格式显示 DATETIME 值。fsp 参数是表示秒精度，取值范围为0-6，默认值取0。
+
+TIMESTAMP[(fsp)]
+> 时间戳。支持的范围是`1970-01-01 00:00:01.000000`到`2038-01-19 03:14:07.999999`。
+fsp 参数是表示秒精度，取值范围为0-6，默认值取0。
+
+TIME[(fsp)]
+> 时间。范围是`-838:59:59.000000`到`838:59:59.000000`。以`HH:MM:SS[.fraction]`格式显示 TIME 值。
+fsp 参数是表示秒精度，取值范围为：0-6。默认值取0。
```
Comment (zimulala) [inline sql/datatype.md]: 数据或者字母连接汉子的时候需要加空格

**Comment by zimulala** [comment]
LGTM

**Pull request merged**